pkg/models/v1/models.go

```diff
@@ -237,14 +237,22 @@ func GetVASPEmailLog(vasp *pb.VASP) (emails []*VASPEmailEntry, err error) {
 	return emails, nil
 }
 
-// TODO: cleanup the duplicated code between this function and contacts.GetSentEmailCount
+// GetSentAdminEmailCount returns the number of emails sent to vasp admin for the given reason.
 func GetSentAdminEmailCount(vasp *pb.VASP, reason string, timeWindowDays int) (sent int, err error) {
 	var adminEmailLog []*EmailLogEntry
 	if adminEmailLog, err = GetAdminEmailLog(vasp); err != nil {
 		return 0, err
 	}
 
-	for _, value := range adminEmailLog {
+	if sent, err = countSentEmails(adminEmailLog, reason, timeWindowDays); err != nil {
+		return 0, err
+	}
```

Comment:
The caller has to check the error anyway so we can just return `return countSentEmails(adminEmailLog, reason, timeWindowDays)` here. Alternatively, since this is looking more like a pass-through method after your refactoring we could probably get rid of this method as described in my other comment.